packages/arb-validator/checkpointing/indexedCheckpointer.go

```diff
@@ -426,44 +348,30 @@ func (cp *IndexedCheckpointer) deleteCheckpointForId(id *common.BlockId) error {
 	return nil
 }
 
-type restoreContextLocked struct {
```

Comment:
Abstracting the restore context preserves implementation flexibility, and supports testing.  For example, if we wanted the restore process to prefetch values, rather than waiting for the client of this interface to ask for them, we would want the client of this code to not be locked into using an IndexedCheckpointer as its restore context.

Reply:
I didn't delete this, I just moved it down lower in the file.